api/src/models/hardware.rs

```diff
@@ -1,23 +1,15 @@
 use crate::schema::hardware_specs;
 use diesel::{
-    dsl, expression::bound::Bound, prelude::*, query_builder::InsertStatement,
-    sql_types::Text, Identifiable, Queryable,
-};
-use gdlk::{
-    validator::{Validate, ValidationErrors},
-    Valid,
+    prelude::*, query_builder::InsertStatement, Identifiable, Queryable,
 };
+use gdlk::{validator::ValidationErrors, Valid};
 use std::convert::TryFrom;
 use uuid::Uuid;
 
-/// Eq clause to filter hardware_specs by slug
-pub type WithSlug<'a> =
-    dsl::Eq<hardware_specs::columns::slug, Bound<Text, &'a str>>;
-
```

Comment:
I removed all these helper types and the SQL helper functions on the model structs. The querying in the API doesn't need any of those queries more than once, so it doesn't really make sense to define helpers for them. Just makes the query harder to read.